common_core.c

```diff
@@ -122,7 +122,7 @@ int ta_find_transactions_by_tag(const iota_client_service_t* const service,
   ret = iota_client_find_transactions(service, find_tx_req, find_tx_res);
 
   if (ret == RC_OK) {
-    ret = hash243_queue_push(&find_tx_res->hashes, res);
+    res->hashes = find_tx_res->hashes;
```

Comment:
Below should not use `find_transactions_res_free` directly, otherwise this will also free `res->hashes`.

We should first `find_tx_res->hashes = NULL` in this block, then we could use `find_transactions_res_free` safely.